lib/rb_shift/client.rb

```diff
@@ -22,30 +22,56 @@ def initialize(url, bearer_token)
       @openshift  = RestClient::Resource.new "#{url}/oapi/v1",
                                              verify_ssl: OpenSSL::SSL::VERIFY_NONE,
                                              headers:    { Authorization: "Bearer #{bearer_token}" }
-      @_projects  = Project.list
     end
 
     # rubocop:disable Metrics/AbcSize
     def get(resource, **opts)
       request = String.new
       request << "namespaces/#{opts[:namespace]}/" if opts[:namespace]
       request << resource.to_s
-      request << opts[:name].to_s if opts[:name]
+      request << "/#{opts[:name].to_s}" if opts[:name]
       client = client resource
-      process_response JSON.parse(client[request].get, :symbolize_names => true)
+      process_response JSON.parse(client[request].get, symbolize_names: true)
+    end
+
+    def read_link(link)
+      v = RestClient.get "#{@url}#{link}", Authorization: "Bearer #{@token}"
+      process_response JSON.parse(v, symbolize_names: true)
     end
 
     def process_response(response)
       return response[:items] if response[:items]
-      return response[:metadata] if response[:metadata]
+      response
     end
 
     def create_project(name, **opts)
-      `oc new-project #{name}`
+      execute "new-project #{name}", **opts
+      project = nil
+      project = projects(true).find { |p| p.name == name } while project.nil?
```

Comment:
I think that because reloading projects is quite time consuming operation, this cycle will in most cases take one iteration. Only when a cluster will be heavily used the operation will take two cycles.